swift/extractor/infra/SwiftMangledName.cpp

```diff
@@ -26,4 +26,30 @@ std::string SwiftMangledName::str() const {
   return out;
 }
 
+SwiftMangledName& SwiftMangledName::operator<<(UntypedTrapLabel label) & {
```

Comment:
I think the `vector<variant<...>>` is overkill, but maybe I'm missing something. I would expect mangled names to be relatively short on average. For short names with not too many components, I'd expect simply appending to a string to be cheaper than a vector+variant. (And indeed since `str()` basically does just that, we actually have the vector+variant machinery *and* a sequence of `StrAppend`s instead of just a sequence of `StrAppend`s.

If I'm missing something and we're looking at huge names, would `absl::Cord` be a simpler but still performant alternative?

Reply:
> I would expect mangled names to be relatively short on average.

As a side note: I don't have enough data to reason about the average, but many names are very very long.

Reply:
because of the id reference mechanism, at least names are not proportional to the nested depth of the declaration. As far as I can tell the outliers are functions with a very, very long list of parameters (and there are a few of those even in the standard library). Still, @sashabu's reasoning is sound: in any case we do string appending at the end, and no other actual operation involving parts, so we can just do it along the way without storing a more complex data structure.